execute/plugin_usdc_test.go

```diff
@@ -0,0 +1,515 @@
+	"github.com/smartcontractkit/chainlink-ccip/internal/mocks"
+	"github.com/smartcontractkit/chainlink-ccip/internal/mocks/inmem"
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugintypes"
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+	mock_types "github.com/smartcontractkit/chainlink-ccip/mocks/execute/exectypes"
+	readermock "github.com/smartcontractkit/chainlink-ccip/mocks/pkg/contractreader"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/contractreader"
+	readerpkg "github.com/smartcontractkit/chainlink-ccip/pkg/reader"
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+	plugintypes2 "github.com/smartcontractkit/chainlink-ccip/plugintypes"
+)
+
+const (
+	address = "0x888888cfaebbed5554c3f36bfbd233f822e9455f"
+)
+
+func Test_USDC_Transfer(t *testing.T) {
+	ctx := context.Background()
+	lggr := logger.Test(t)
+
+	sourceChain := cciptypes.ChainSelector(sel.ETHEREUM_TESTNET_SEPOLIA.Selector)
+	destChain := cciptypes.ChainSelector(sel.ETHEREUM_MAINNET_BASE_1.Selector)
+
+	nodesSetup, server := setupSimpleTest(ctx, t, lggr, sourceChain, destChain)
+	defer server.Close()
+
+	nodes := make([]ocr3types.ReportingPlugin[[]byte], 0, len(nodesSetup))
+	for _, n := range nodesSetup {
+		nodes = append(nodes, n.node)
+	}
+
+	nodeIDs := make([]commontypes.OracleID, 0, len(nodesSetup))
+	for _, n := range nodesSetup {
+		nodeIDs = append(nodeIDs, n.node.ReportingCfg().OracleID)
+	}
+
+	runner := testhelpers.NewOCR3Runner(nodes, nodeIDs, nil)
+
+	// Contract Discovery round.
+	res, err := runner.RunRound(ctx)
+	require.NoError(t, err)
+	outcome, err := exectypes.DecodeOutcome(res.Outcome)
+	require.NoError(t, err)
+	require.Equal(t, exectypes.Initialized, outcome.State)
+
+	// Round 1 - Get Commit Reports
+	// One pending commit report only.
+	// Two of the messages are executed which should be indicated in the Outcome.
+	res, err = runner.RunRound(ctx)
+	require.NoError(t, err)
+	outcome, err = exectypes.DecodeOutcome(res.Outcome)
+	require.NoError(t, err)
+	require.Len(t, outcome.Report.ChainReports, 0)
+	require.Len(t, outcome.PendingCommitReports, 1)
+	require.ElementsMatch(t, outcome.PendingCommitReports[0].ExecutedMessages, []cciptypes.SeqNum{100, 101})
+
+	// Round 2 - Get Messages
+	// Messages now attached to the pending commit.
+	res, err = runner.RunRound(ctx)
+	require.NoError(t, err)
+	outcome, err = exectypes.DecodeOutcome(res.Outcome)
+	require.NoError(t, err)
+	require.Len(t, outcome.Report.ChainReports, 0)
+	require.Len(t, outcome.PendingCommitReports, 1)
+
+	// Round 3 - Filter
+	// An execute report with the following messages executed: 102, 103, 104, 105.
+	res, err = runner.RunRound(ctx)
+	require.NoError(t, err)
+	outcome, err = exectypes.DecodeOutcome(res.Outcome)
+	require.NoError(t, err)
+	sequenceNumbers := slicelib.Map(outcome.Report.ChainReports[0].Messages, func(m cciptypes.Message) cciptypes.SeqNum {
```

Comment:
I really wish such basic utilities are part of https://pkg.go.dev/golang.org/x/exp/slices :-(